src/main/java/com.careem.opensource/Parser.java

```diff
@@ -49,9 +63,26 @@ public boolean shouldReadMoreLine(String line) {
   private double parseDecimalNumber(String line) {
     Matcher matcher = DECIMAL_NUMBER_PATTERN.matcher(line);
     if (matcher.find()) {
-      return Double.valueOf(matcher.group(0));
+      return Double.valueOf(matcher.group(matcher.groupCount()-1));
     } else {
       return 0.0;
     }
   }
+  private String extractSecs(String chunk){
+    Matcher matcher = EVACUATION_PAUSE_PATTERN.matcher(chunk);
+    if (matcher.find()) {
+      return matcher.group(0);
+    } else {
+      return "1.21 secs]";
+    }
+  }
+  private void analyzeConcurrentMark(GcData.GcDataBuilder gcDataBuilder,String chunk){
```

Comment:
there should be a space after comma. this should be corrected when you autoformat your code